aws/logs_monitoring/steps/enums.py

```diff
@@ -0,0 +1,176 @@
+    # e.g. AWSLogs/123456779121/vpcflowlogs/us-east-1/2020/10/02/123456779121_vpcflowlogs_us-east-1_fl-xxxxx.log.gz
+    VPC = "vpcflowlogs"
+    # e.g. 2020/10/02/21/aws-waf-logs-testing-1-2020-10-02-21-25-30-x123x-x456x or AWSLogs/123456779121/WAFLogs/us-east-1/xxxxxx-waf/2022/10/11/14/10/123456779121_waflogs_us-east-1_xxxxx-waf_20221011T1410Z_12756524.log.gz
+    WAF_0 = "aws-waf-logs"
+    WAF_1 = "waflogs"
+
+    def __str__(self):
+        return f"{self.value}"
+
+
+KEYWORD_TO_SOURCE_MAP = {
+    AwsS3EventSourceKeyword.BEDROCK: AwsEventSource.BEDROCK,
+    AwsS3EventSourceKeyword.CARBONBLACK: AwsEventSource.CARBONBLACK,
+    AwsS3EventSourceKeyword.CODEBUILD: AwsEventSource.CODEBUILD,
+    AwsS3EventSourceKeyword.CLOUDFRONT: AwsEventSource.CLOUDFRONT,
+    AwsS3EventSourceKeyword.DMS: AwsEventSource.DMS,
+    AwsS3EventSourceKeyword.DOCDB: AwsEventSource.DOCDB,
+    AwsS3EventSourceKeyword.ELB: AwsEventSource.ELB,
+    AwsS3EventSourceKeyword.KINESIS: AwsEventSource.KINESIS,
+    AwsS3EventSourceKeyword.MSK: AwsEventSource.MSK,
+    AwsS3EventSourceKeyword.NETWORKFIREWALL: AwsEventSource.NETWORKFIREWALL,
+    AwsS3EventSourceKeyword.REDSHIFT: AwsEventSource.REDSHIFT,
+    AwsS3EventSourceKeyword.ROUTE53: AwsEventSource.ROUTE53,
+    AwsS3EventSourceKeyword.TRANSITAGATEWAY: AwsEventSource.TRANSITGATEWAY,
+    AwsS3EventSourceKeyword.VERIFIED_ACCESS: AwsEventSource.VERIFIED_ACCESS,
+    AwsS3EventSourceKeyword.VPC: AwsEventSource.VPC,
+    AwsS3EventSourceKeyword.WAF_0: AwsEventSource.WAF,
+    AwsS3EventSourceKeyword.WAF_1: AwsEventSource.WAF,
+}
+
+
+class AwsCwEventSourcePrefix(Enum):
+    # default location for rest api execution logs
+    APIGATEWAY_0 = "api-gateway"
+    # default location set by serverless framework for rest api access logs
+    APIGATEWAY_1 = "/aws/api-gateway"
+    # default location set by serverless framework for http api logs
+    APIGATEWAY_2 = "/aws/http-api"
+    # WebSocket API Execution Logs, e.g. /aws/apigateway/api-id/stage-name
+    APIGATEWAY_3 = "/aws/apigateway"
+    # e.g. /aws/appsync/yourApiId
+    APPSYNC = "/aws/appsync"
+    BEDROCK = "aws/bedrock/modelinvocations"
+    # e.g. /aws/codebuild/my-project
+    CODEBUILD = "/aws/codebuild"
+    CLOUDTRAIL = "_CloudTrail_"
+    # e.g. dms-tasks-test-instance
+    DMS = "dms-tasks"
+    # e.g. /aws/docdb/yourClusterName/profile
+    DOCDB = "/aws/docdb"
+    # e.g. /aws/eks/yourClusterName/profile
+    EKS = "/aws/eks"
+    # e.g. /aws/fsx/windows/xxx
+    FSX = "/aws/fsx/windows"
+    # e.g. /aws/kinesisfirehose/dev
+    KINESIS = "/aws/kinesis"
+    # e.g. /aws/lambda/helloDatadog
+    lAMBDA = "/aws/lambda"
+    RDS = "/aws/rds"
+    # e.g. sns/us-east-1/123456779121/SnsTopicX
+    SNS = "sns/"
+    STEPFUNCTION = "/aws/vendedlogs/states"
+    TRANSITGATEWAY = "tgw-attach"
+
+    def __str__(self):
+        return f"{self.value}"
+
+
+PREFIX_TO_SOURCE_MAP = {
+    AwsCwEventSourcePrefix.APIGATEWAY_0: AwsEventSource.APIGATEWAY,
```

Comment:
Could these maps be defined in the opposite way to avoid repetition ? (could be done in another PR)
sth like : 

```python
MAP = {
    AwsEventSource.APIGATEWAY : [
      AwsCwEventSourcePrefix.APIGATEWAY_0,  
      AwsCwEventSourcePrefix.APIGATEWAY_1,
      AwsCwEventSourcePrefix.APIGATEWAY_2,
      AwsCwEventSourcePrefix.APIGATEWAY_3
  ],
   AwsEventSource.APPSYNC:  AwsCwEventSourcePrefix.APPSYNC,
//...
```

Reply:
I'll see if I can do it this way, the thing is in one handler we check the prefix in another handler we check the keyword

Reply:
getting back on this, I prefer to have the prefix map separate from the keyword map. Also, having the prefix in the key of the map is faster for lookups